### DISCUSSION TITLE
minio fail to start

### DISCUSSION THREAD

**Discussion by liujiaxiang** [minio/minio#13466] [Q&A] [answered]
version:minio version RELEASE.2020-07-02T00-15-09Z
start command:./minio server /minio_data/data1 /minio_data/data2 /minio_data/data3 /minio_data/data4
description:It was normal before，after restarting the server, it will not start normally.
logs:
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/minio/minio/cmd.(*xlSets).getHashedSet(0xc000764640, 0x1c10721, 0x17, 0x40c2d6)
	github.com/minio/minio@/cmd/xl-sets.go:600 +0x9f
github.com/minio/minio/cmd.(*xlSets).NewNSLock(0xc000764640, 0x1fb2fc0, 0xc000212e40, 0x1bf791e, 0xa, 0xc0006624f0, 0x1, 0x1, 0xc000212e01, 0xc0006624f0)
	github.com/minio/minio@/cmd/xl-sets.go:369 +0x6b
github.com/minio/minio/cmd.(*xlZones).NewNSLock(0xc0006ab180, 0x1fb2fc0, 0xc000212e40, 0x1bf791e, 0xa, 0xc0006624f0, 0x1, 0x1, 0x48, 0x50)
	github.com/minio/minio@/cmd/xl-zones.go:86 +0x94
github.com/minio/minio/cmd.initSafeMode(0x1fb2fc0, 0xc0002a4240, 0x1ff7580, 0xc0006ab180, 0x0, 0x0)
	github.com/minio/minio@/cmd/server-main.go:184 +0x134
github.com/minio/minio/cmd.serverMain(0xc0000bab60)
	github.com/minio/minio@/cmd/server-main.go:512 +0xb5e
github.com/minio/cli.HandleAction(0x1894500, 0x1c839b8, 0xc0000bab60, 0xc0000b6900, 0x0)
	github.com/minio/cli@v1.22.0/app.go:500 +0x7c
github.com/minio/cli.Command.Run(0x1bf29c0, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1c1a6fb, 0x1b, 0x0, ...)
	github.com/minio/cli@v1.22.0/command.go:225 +0x991
github.com/minio/cli.(*App).Run(0xc0000ba1a0, 0xc0000b6120, 0x6, 0x6, 0x0, 0x0)
	github.com/minio/cli@v1.22.0/app.go:261 +0x6ab
github.com/minio/minio/cmd.Main(0xc0000b6120, 0x6, 0x6)
	github.com/minio/minio@/cmd/main.go:169 +0x89
main.main()
	github.com/minio/minio@/main.go:37 +0x45

**Comment by harshavardhana** [comment, marked as answer]
> version:minio version RELEASE.2020-07-02T00-15-09Z
> start command:./minio server /minio_data/data1 /minio_data/data2 /minio_data/data3 /minio_data/data4
> description:It was normal before，after restarting the server, it will not start normally.
> logs:
> panic: runtime error: index out of range [-1]

You seem to be running older MinIO on backend disks with newer data format.

**Reply by liujiaxiang** [reply]
But it worked normally before

**Reply by liujiaxiang** [reply]
I change to use minio.RELEASE.2020-11-06T23-17-07Z，Is there something wrong with erasure code
logs:
panic: runtime error: index out of range [-1]

goroutine 114 [running]:
github.com/minio/minio/cmd.(*erasureSets).getHashedSet(...)
	github.com/minio/minio/cmd/erasure-sets.go:601
github.com/minio/minio/cmd.(*erasureSets).NewNSLock(0xc0009c1b80, 0x1ddd905, 0xa, 0xc0000a57e0, 0x1, 0x1, 0x1, 0xc0000a57e0)
	github.com/minio/minio/cmd/erasure-sets.go:414 +0x2a5
github.com/minio/minio/cmd.(*erasureServerSets).NewNSLock(0xc000351f00, 0x1ddd905, 0xa, 0xc0000a57e0, 0x1, 0x1, 0xc0004cf2a0, 0x1e74aa8)
	github.com/minio/minio/cmd/erasure-server-sets.go:92 +0x70
github.com/minio/minio/cmd.runDataCrawler(0x2114720, 0xc0002f2040, 0x213cf80, 0xc000351f00)
	github.com/minio/minio/cmd/data-crawler.go:72 +0xa5
created by github.com/minio/minio/cmd.initDataCrawler
	github.com/minio/minio/cmd/data-crawler.go:63 +0xad

**Reply by harshavardhana** [reply]
yes because you have drives that have upgraded `format.json`  you cannot use the same drives that were used by the newer version of MinIO to be used by older ones.

**Reply by liujiaxiang** [reply]
Which version should I use.
I change to use minio.RELEASE.2021-06-07T21-40-51Z.
the logs:
API: SYSTEM()
Time: 11:26:21 CST 10/19/2021
DeploymentID: 104ce702-ecbf-4d7e-a988-30e01356d10e
Error: Unable to initialize sub-systems: Unable to migrate all config at .minio.sys/config/: Decrypting config failed sio: data is not authentic, possibly credentials are incorrect (*fmt.wrapError)
       1: cmd/server-main.go:548:cmd.serverMain()

API: SYSTEM()
Time: 11:26:21 CST 10/19/2021
DeploymentID: 104ce702-ecbf-4d7e-a988-30e01356d10e
Error: Unable to migrate IAM users and policies to new format: cmd.iamFormat.readFieldHash: expect ", but found f, error found in #2 byte of ...|{fv񅱢x
                                                                                                                                                      鮼..., bigger context ...|{fv񅱢x
         鮜®Lz·S%b¨:ي𪺥ߔαµ@b§3̖ą<... (*fmt.wrapError)
       1: cmd/iam.go:647:cmd.(*IAMSys).Init()

API: SYSTEM()
Time: 11:26:21 CST 10/19/2021
DeploymentID: 104ce702-ecbf-4d7e-a988-30e01356d10e
Error: IAM sub-system is partially initialized, some users may not be available (*errors.errorString)
       1: cmd/iam.go:648:cmd.(*IAMSys).Init()
Status:         4 Online, 0 Offline. 
Endpoint: http://10.2.100.18:9000  http://127.0.0.1:9000     

Browser Access:
   http://10.2.100.18:9000  http://127.0.0.1:9000    

Object API (Amazon S3 compatible):
   Go:         https://docs.min.io/docs/golang-client-quickstart-guide
   Java:       https://docs.min.io/docs/java-client-quickstart-guide
   Python:     https://docs.min.io/docs/python-client-quickstart-guide
   JavaScript: https://docs.min.io/docs/javascript-client-quickstart-guide
   .NET:       https://docs.min.io/docs/dotnet-client-quickstart-guide

API: SYSTEM()
Time: 11:27:21 CST 10/19/2021
DeploymentID: 104ce702-ecbf-4d7e-a988-30e01356d10e
Error: dataUsageCache: unknown version: 6 (*errors.errorString)
       5: internal/logger/logonce.go:54:logger.(*logOnceType).logOnceIf()
       4: internal/logger/logonce.go:94:logger.LogOnceIf()
       3: cmd/data-usage-cache.go:705:cmd.(*dataUsageCache).load()
       2: cmd/erasure.go:371:cmd.erasureObjects.nsScanner()
       1: cmd/erasure-server-pool.go:440:cmd.(*erasureServerPools).NSScanner.func2()

**Reply by klauspost** [reply]
`Error: dataUsageCache: unknown version: 6 (*errors.errorString)` indicates that you have been running a newer version than the current. For this particular functionality it isn't a big problem (it should recover), but I am not sure about IAM.

TBH your system seems quite messed up after several down+upgrades, so I am not sure how to recover easily. Maybe @harshavardhana has a manual way to reset IAM so you can start from scratch with that.
